[iOS] Migrating to 4.1.0 throws copy error with Headers (.h)

**Description:**

Migrated from 3.1.6 to latest 4.1.0. 
Now when building the project I face an odd issue. Tried of course to rebuild, clear bin&obj etc ... 

> 5>C:\Users\casta\.nuget\packages\onesignalsdk.xamarin\4.1.0\build\Xamarin.iOS\OneSignalSDK.Xamarin.targets(6,9): warning MSB3026: Could not copy "C:\Users\casta\.nuget\packages\onesignalsdk.xamarin\4.1.0\build\Xamarin.iOS\..\..\content\OneSignalSDK.Xamarin.iOS.resources\OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h" to "D:\DEVELOPPEMENT\MP\PrivateFront\Private\Private.iOS\bin\iPhone\Debug\/OneSignalSDK.Xamarin.iOS.resources/OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h". Beginning retry 1 in 1000ms. Could not find a part of the path 'D:\DEVELOPPEMENT\MP\PrivateFront\Private\Private.iOS\bin\iPhone\Debug\/OneSignalSDK.Xamarin.iOS.resources/OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h'. 
5>C:\Users\casta\.nuget\packages\onesignalsdk.xamarin\4.1.0\build\Xamarin.iOS\OneSignalSDK.Xamarin.targets(6,9): warning MSB3026: Could not copy "C:\Users\casta\.nuget\packages\onesignalsdk.xamarin\4.1.0\build\Xamarin.iOS\..\..\content\OneSignalSDK.Xamarin.iOS.resources\OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h" to "D:\DEVELOPPEMENT\MP\PrivateFront\Private\Private.iOS\bin\iPhone\Debug\/OneSignalSDK.Xamarin.iOS.resources/OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h". Beginning retry 2 in 1000ms. Could not find a part of the path 'D:\DEVELOPPEMENT\MP\PrivateFront\Private\Private.iOS\bin\iPhone\Debug\/OneSignalSDK.Xamarin.iOS.resources/OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h'. 
5>C:\Users\casta\.nuget\packages\onesignalsdk.xamarin\4.1.0\build\Xamarin.iOS\OneSignalSDK.Xamarin.targets(6,9): warning MSB3026: Could not copy "C:\Users\casta\.nuget\packages\onesignalsdk.xamarin\4.1.0\build\Xamarin.iOS\..\..\content\OneSignalSDK.Xamarin.iOS.resources\OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h" to "D:\DEVELOPPEMENT\MP\PrivateFront\Private\Private.iOS\bin\iPhone\Debug\/OneSignalSDK.Xamarin.iOS.resources/OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h". Beginning retry 3 in 1000ms. Could not find a part of the path 'D:\DEVELOPPEMENT\MP\PrivateFront\Private\Private.iOS\bin\iPhone\Debug\/OneSignalSDK.Xamarin.iOS.resources/OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h'. 
5>C:\Users\casta\.nuget\packages\onesignalsdk.xamarin\4.1.0\build\Xamarin.iOS\OneSignalSDK.Xamarin.targets(6,9): warning MSB3026: Could not copy "C:\Users\casta\.nuget\packages\onesignalsdk.xamarin\4.1.0\build\Xamarin.iOS\..\..\content\OneSignalSDK.Xamarin.iOS.resources\OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h" to "D:\DEVELOPPEMENT\MP\PrivateFront\Private\Private.iOS\bin\iPhone\Debug\/OneSignalSDK.Xamarin.iOS.resources/OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h". Beginning retry 4 in 1000ms. Could not find a part of the path 'D:\DEVELOPPEMENT\MP\PrivateFront\Private\Private.iOS\bin\iPhone\Debug\/OneSignalSDK.Xamarin.iOS.resources/OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h'. 
5>C:\Users\casta\.nuget\packages\onesignalsdk.xamarin\4.1.0\build\Xamarin.iOS\OneSignalSDK.Xamarin.targets(6,9): warning MSB3026: Could not copy "C:\Users\casta\.nuget\packages\onesignalsdk.xamarin\4.1.0\build\Xamarin.iOS\..\..\content\OneSignalSDK.Xamarin.iOS.resources\OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h" to "D:\DEVELOPPEMENT\MP\PrivateFront\Private\Private.iOS\bin\iPhone\Debug\/OneSignalSDK.Xamarin.iOS.resources/OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h". Beginning retry 5 in 1000ms. Could not find a part of the path 'D:\DEVELOPPEMENT\MP\PrivateFront\Private\Private.iOS\bin\iPhone\Debug\/OneSignalSDK.Xamarin.iOS.resources/OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h'. 
5>C:\Users\casta\.nuget\packages\onesignalsdk.xamarin\4.1.0\build\Xamarin.iOS\OneSignalSDK.Xamarin.targets(6,9): warning MSB3026: Could not copy "C:\Users\casta\.nuget\packages\onesignalsdk.xamarin\4.1.0\build\Xamarin.iOS\..\..\content\OneSignalSDK.Xamarin.iOS.resources\OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h" to "D:\DEVELOPPEMENT\MP\PrivateFront\Private\Private.iOS\bin\iPhone\Debug\/OneSignalSDK.Xamarin.iOS.resources/OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h". Beginning retry 6 in 1000ms. Could not find a part of the path 'D:\DEVELOPPEMENT\MP\PrivateFront\Private\Private.iOS\bin\iPhone\Debug\/OneSignalSDK.Xamarin.iOS.resources/OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h'. 
5>C:\Users\casta\.nuget\packages\onesignalsdk.xamarin\4.1.0\build\Xamarin.iOS\OneSignalSDK.Xamarin.targets(6,9): warning MSB3026: Could not copy "C:\Users\casta\.nuget\packages\onesignalsdk.xamarin\4.1.0\build\Xamarin.iOS\..\..\content\OneSignalSDK.Xamarin.iOS.resources\OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h" to "D:\DEVELOPPEMENT\MP\PrivateFront\Private\Private.iOS\bin\iPhone\Debug\/OneSignalSDK.Xamarin.iOS.resources/OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h". Beginning retry 7 in 1000ms. Could not find a part of the path 'D:\DEVELOPPEMENT\MP\PrivateFront\Private\Private.iOS\bin\iPhone\Debug\/OneSignalSDK.Xamarin.iOS.resources/OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h'. 
5>C:\Users\casta\.nuget\packages\onesignalsdk.xamarin\4.1.0\build\Xamarin.iOS\OneSignalSDK.Xamarin.targets(6,9): warning MSB3026: Could not copy "C:\Users\casta\.nuget\packages\onesignalsdk.xamarin\4.1.0\build\Xamarin.iOS\..\..\content\OneSignalSDK.Xamarin.iOS.resources\OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h" to "D:\DEVELOPPEMENT\MP\PrivateFront\Private\Private.iOS\bin\iPhone\Debug\/OneSignalSDK.Xamarin.iOS.resources/OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h". Beginning retry 8 in 1000ms. Could not find a part of the path 'D:\DEVELOPPEMENT\MP\PrivateFront\Private\Private.iOS\bin\iPhone\Debug\/OneSignalSDK.Xamarin.iOS.resources/OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h'. 
5>C:\Users\casta\.nuget\packages\onesignalsdk.xamarin\4.1.0\build\Xamarin.iOS\OneSignalSDK.Xamarin.targets(6,9): warning MSB3026: Could not copy "C:\Users\casta\.nuget\packages\onesignalsdk.xamarin\4.1.0\build\Xamarin.iOS\..\..\content\OneSignalSDK.Xamarin.iOS.resources\OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h" to "D:\DEVELOPPEMENT\MP\PrivateFront\Private\Private.iOS\bin\iPhone\Debug\/OneSignalSDK.Xamarin.iOS.resources/OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h". Beginning retry 9 in 1000ms. Could not find a part of the path 'D:\DEVELOPPEMENT\MP\PrivateFront\Private\Private.iOS\bin\iPhone\Debug\/OneSignalSDK.Xamarin.iOS.resources/OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h'. 
5>C:\Users\casta\.nuget\packages\onesignalsdk.xamarin\4.1.0\build\Xamarin.iOS\OneSignalSDK.Xamarin.targets(6,9): warning MSB3026: Could not copy "C:\Users\casta\.nuget\packages\onesignalsdk.xamarin\4.1.0\build\Xamarin.iOS\..\..\content\OneSignalSDK.Xamarin.iOS.resources\OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h" to "D:\DEVELOPPEMENT\MP\PrivateFront\Private\Private.iOS\bin\iPhone\Debug\/OneSignalSDK.Xamarin.iOS.resources/OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h". Beginning retry 10 in 1000ms. Could not find a part of the path 'D:\DEVELOPPEMENT\MP\PrivateFront\Private\Private.iOS\bin\iPhone\Debug\/OneSignalSDK.Xamarin.iOS.resources/OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h'. 
5>C:\Users\casta\.nuget\packages\onesignalsdk.xamarin\4.1.0\build\Xamarin.iOS\OneSignalSDK.Xamarin.targets(6,9): error MSB3027: Could not copy "C:\Users\casta\.nuget\packages\onesignalsdk.xamarin\4.1.0\build\Xamarin.iOS\..\..\content\OneSignalSDK.Xamarin.iOS.resources\OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h" to "D:\DEVELOPPEMENT\MP\PrivateFront\Private\Private.iOS\bin\iPhone\Debug\/OneSignalSDK.Xamarin.iOS.resources/OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h". Exceeded retry count of 10. Failed. 
5>C:\Users\casta\.nuget\packages\onesignalsdk.xamarin\4.1.0\build\Xamarin.iOS\OneSignalSDK.Xamarin.targets(6,9): error MSB3021: Unable to copy file "C:\Users\casta\.nuget\packages\onesignalsdk.xamarin\4.1.0\build\Xamarin.iOS\..\..\content\OneSignalSDK.Xamarin.iOS.resources\OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h" to "D:\DEVELOPPEMENT\MP\PrivateFront\Private\Private.iOS\bin\iPhone\Debug\/OneSignalSDK.Xamarin.iOS.resources/OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h". Could not find a part of the path 'D:\DEVELOPPEMENT\MP\PrivateFront\Private\Private.iOS\bin\iPhone\Debug\/OneSignalSDK.Xamarin.iOS.resources/OneSignalExtension.xcframework\ios-arm64_i386_x86_64-simulator\OneSignalExtension.framework\Headers\OneSignalNotificationServiceExtensionHandler.h'.

**Environment**

OneSignalSDK : 4.1.0
Xamarin Forms : 5.0
Visual Studio : 2022 - 17.2.5 
Target Framework : iOS 15.4 


**Steps to Reproduce Issue:**

1. Use VS 17.2.5 
2. Use Xamarin SDK 4.1.0
3. Build project . 

**Anything else:**

(crash stacktraces, as well as any other information here)

--- a/OneSignal.iOS.Binding/OneSignalSDK.Xamarin.targets
+++ b/OneSignal.iOS.Binding/OneSignalSDK.Xamarin.targets
@@ -1,7 +1,7 @@
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
     <Target Name="OneSignalAddOneSignalXCFramework" BeforeTargets="BeforeCompile">
          <PropertyGroup>
-            <BindingResources>$(MSBuildThisFileDirectory)../../content/OneSignalSDK.Xamarin.iOS.resources</BindingResources>
+            <BindingResources>$(MSBuildThisFileDirectory)../../res/iOS</BindingResources>
         </PropertyGroup>
         <ItemGroup>
             <NativeReference Include="$(BindingResources)/OneSignal.xcframework">

--- a/OneSignal.iOS.Binding/OneSignalSDK.Xamarin.targets
+++ b/OneSignal.iOS.Binding/OneSignalSDK.Xamarin.targets
@@ -1,29 +1,28 @@
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
     <Target Name="OneSignalAddOneSignalXCFramework" BeforeTargets="BeforeCompile">
+         <PropertyGroup>
+            <BindingResources>$(MSBuildThisFileDirectory)../../content/OneSignalSDK.Xamarin.iOS.resources</BindingResources>
+        </PropertyGroup>
         <ItemGroup>
-            <BindingResources Include="$(MSBuildThisFileDirectory)../../content/OneSignalSDK.Xamarin.iOS.resources/**/*.*" />
-        </ItemGroup>
-        <Copy SourceFiles="@(BindingResources)" DestinationFolder="$(TargetDir)/OneSignalSDK.Xamarin.iOS.resources/%(RecursiveDir)" ContinueOnError="false" />
-        <ItemGroup>
-            <NativeReference Include="$(TargetDir)/OneSignalSDK.Xamarin.iOS.resources\OneSignal.xcframework">
+            <NativeReference Include="$(BindingResources)/OneSignal.xcframework">
                 <Kind>Framework</Kind>
                 <SmartLink>False</SmartLink>
                 <ForceLoad>True</ForceLoad>
                 <Frameworks>SystemConfiguration UserNotifications WebKit CoreGraphics UIKit OneSignalOutcomes OneSignalCore OneSignalExtension</Frameworks>
             </NativeReference>
-            <NativeReference Include="$(TargetDir)/OneSignalSDK.Xamarin.iOS.resources\OneSignalExtension.xcframework">
+            <NativeReference Include="$(BindingResources)/OneSignalExtension.xcframework">
                 <Kind>Framework</Kind>
                 <SmartLink>False</SmartLink>
                 <ForceLoad>True</ForceLoad>
                 <Frameworks>UserNotifications OneSignalCore OneSignalOutcomes</Frameworks>
             </NativeReference>
-            <NativeReference Include="$(TargetDir)/OneSignalSDK.Xamarin.iOS.resources\OneSignalOutcomes.xcframework">
+            <NativeReference Include="$(BindingResources)/OneSignalOutcomes.xcframework">
                 <Kind>Framework</Kind>
                 <SmartLink>False</SmartLink>
                 <ForceLoad>True</ForceLoad>
                 <Frameworks>OneSignalCore</Frameworks>
             </NativeReference>
-            <NativeReference Include="$(TargetDir)/OneSignalSDK.Xamarin.iOS.resources\OneSignalCore.xcframework">
+            <NativeReference Include="$(BindingResources)/OneSignalCore.xcframework">
                 <Kind>Framework</Kind>
                 <SmartLink>False</SmartLink>
                 <ForceLoad>True</ForceLoad>

--- a/OneSignalSDK.Xamarin.nuspec
+++ b/OneSignalSDK.Xamarin.nuspec
@@ -55,7 +55,7 @@
         <!-- Workaround to support .XCFramework for iOS  -->
         <!-- Resources includes the full OneSignal.XCFramework.
              iOS.Binding project has NoBindingEmbedding = true so it isn't also bundled in the .dll -->
-        <file src="OneSignal.iOS.Binding\bin\Release\OneSignal.iOS.Binding.resources\**" target="content\OneSignalSDK.Xamarin.iOS.resources" />
+        <file src="OneSignal.iOS.Binding\bin\Release\OneSignal.iOS.Binding.resources\**" target="res\iOS" />
         <!-- This is a .target files that gets used by project that consumes the NuGet package.
              This copies out the OneSignal.xcframework from the resources folder and adds a NativeReference to it in the app project. -->
         <file src="OneSignal.iOS.Binding\OneSignalSDK.Xamarin.targets" target="build\Xamarin.iOS\" />

